src/command/defaults/ExperienceCommand.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\command\defaults;
+
+use pocketmine\command\CommandSender;
+use pocketmine\command\utils\InvalidCommandSyntaxException;
+use pocketmine\entity\Attribute;
+use pocketmine\permission\DefaultPermissionNames;
+use pocketmine\utils\AssumptionFailedError;
+use pocketmine\utils\Limits;
+use pocketmine\utils\TextFormat;
+use function abs;
+use function count;
+use function str_ends_with;
+use function substr;
+
+class ExperienceCommand extends VanillaCommand{
+
+	public function __construct(){
+		parent::__construct(
+			"xp",
+			"Adds or removes player experience",
+			"/xp <experience[L]> [player]"
+		);
+		$this->setPermissions([
+			DefaultPermissionNames::COMMAND_XP_SELF,
+			DefaultPermissionNames::COMMAND_XP_OTHER
+		]);
+	}
+
+	public function execute(CommandSender $sender, string $commandLabel, array $args){
+		if(count($args) < 1){
+			throw new InvalidCommandSyntaxException();
+		}
+		$player = $this->fetchPermittedPlayerTarget($sender, $args[1] ?? null, DefaultPermissionNames::COMMAND_XP_SELF, DefaultPermissionNames::COMMAND_XP_OTHER);
+		if($player === null){
+			return true;
+		}
+		$xpManager = $player->getXpManager();
+		if(str_ends_with($args[0], "L")){
+			$xpLevelAttr = $player->getAttributeMap()->get(Attribute::EXPERIENCE_LEVEL) ?? throw new AssumptionFailedError();
```

Comment:
```suggestion
			/** @var Attribute $xpLevelAttr */
			$xpLevelAttr = $player->getAttributeMap()->get(Attribute::EXPERIENCE_LEVEL);
```

Reply:
IIRC, @var(s) in PM code shouldn’t be used for such assertions. Also, I’ve found similar assertion via `AssumptionFailedError` in the code for another attribute that should always exist by default too (https://github.com/pmmp/PocketMine-MP/blob/stable/src/network/mcpe/handler/InGamePacketHandler.php#L510)

Reply:
Yep, @var should only be used if the IDE can't figure out what's going on. AssumptionFailedError is correct here.